source/opt/eliminate_dead_input_components_pass.cpp

```diff
@@ -62,6 +66,14 @@ Pass::Status EliminateDeadInputComponentsPass::Process() {
     }
     const analysis::Array* arr_type = ptr_type->pointee_type()->AsArray();
     if (arr_type != nullptr) {
+      // Only process array if input of vertex shader, or output of
+      // fragment shader. Otherwise, if one shader has a runtime index and the
+      // other does not, interface incompatibility can occur.
+      if (!((ptr_type->storage_class() == spv::StorageClass::Input &&
```

Comment:
Seems like it's safe to remove fragment outputs too, so shall this pass be extended to fragment too (above condition)?
If yes, `vertex_shader_only` could be renamed into `boundary_stages_only` or `ignore_internal_stages` or `ignore_intermediate_stages`?

Reply:
Ok. Added #4987 to track this request.